bigc/exceptions.py

```diff
@@ -1,6 +1,36 @@
+from __future__ import annotations
+
+from typing import Any, Optional
+
+from requests import Response
+import http
+
+
 class BigCommerceAPIException(Exception):
     """Base exception class for BigCommerce API errors"""
 
+    def __init__(self, message: Optional[str] = None, response: Optional[Response] = None):
+        self._message = message
+        self.response = response
+
+        super().__init__(
+            message or f'Request to BigCommerce failed{f" with {self.message}" if response is not None else ""}'
+        )
+
+    @property
+    def status_code(self) -> int:
+        if self.response is None:
+            raise RuntimeError('There is no response for this exception')
+
+        return self.response.status_code
```

Comment:
I'd feel better if this was `Optional[int]` so type checkers would tell you to check nulls instead of needing a `try`/`except`.

Reply:
https://github.com/MedShift/bigc/pull/35/commits/a51e8fe739583ebec7d232a7039deeedeada5555